src/middleware.js

```diff
@@ -2,15 +2,12 @@ import RSAA from './RSAA';
 import { isRSAA, validateRSAA } from './validation';
 import { InvalidRSAA, RequestError } from './errors';
 import { normalizeTypeDescriptors, actionWith } from './util';
+import defaults from './defaults';
 
-/**
- * A Redux middleware that processes RSAA actions.
- *
- * @type {ReduxMiddleware}
- * @access public
- */
-function apiMiddleware({ getState }) {
-  return next => action => {
+function createMiddleware(options = {}) {
```

Comment:
We should document this and how to use it 📚 

Would configuring defaults for your middleware be like:

```js
const apiMiddlewareWithDefaults = createMiddleware({
  fetch: myFetch,
  ok: myOk
})
```
when configuring your store? What other defaults can you set this way?

And then you can override from individual `RSAA`s?

```js
{
  [RSAA]: {
    ...
    ok: myOtherOk
    ...
  }
}
```